### PR TITLE
Backporting upgrade changes to 1.3 release

### PR DESCRIPTION
**Pull request by rofinn**
NOTE: I'll just be making the tag off 1.3 w/o merging. This is just to show a diff between the two releases.

### CODE REVIEW

File: docs/src/upgrading.md [hunk 1]
```diff
@@ -0,0 +1,104 @@
+# Upgrading
+
+JLSO.jl will automatically upgrade older versions of the file format when you call `JLSO.load` or `read`.
```
Comment (omus) [inline docs/src/upgrading.md]: Seems dangerous. This is modifying the file on disk automatically?

Comment (rofinn) [reply docs/src/upgrading.md]: No, `JLSO.load` and `read` will only upgrade the in memory. The purpose of `JLSO.upgrade` is to explicit upgrade the files on disk to make future loading or loading of batches of files faster. The only way to overwrite existing files it to call `JLSO.upgrade` and pass the same path for both `src` and `dest`.

Comment (oxinabox) [reply docs/src/upgrading.md]: It should probably not say "upgrade" for when it loads a old version.
not on the docs at least.

Comment (rofinn) [reply docs/src/upgrading.md]: That's what it says in the logs though.

Comment (omus) [reply docs/src/upgrading.md]: Sounds good to me!

File: Project.toml [hunk 2]
```diff
@@ -2,7 +2,7 @@ name = "JLSO"
 uuid = "9da8a3cd-07a3-59c0-a743-3fdc52c30d11"
 license = "MIT"
 authors = ["Invenia Technical Computing Corperation"]
-version = "1.2.2"
+version = "1.3.0"
 
 [deps]
```
Comment (rofinn) [inline Project.toml]: @JuliaRegistrator register

Comment (oxinabox) [reply Project.toml]: I don't think that is a thing

**Pull request closed without merging**